src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -303,6 +304,15 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			if (contents) {
 				this._defaultLanguageInfo = contents.metadata && contents.metadata.language_info;
 				this._savedKernelInfo = this.getSavedKernelInfo(contents);
+				if (contents.metadata) {
+					Object.keys(contents.metadata).forEach(key => {
+						let expectedKeys = ['kernelspec', 'language_info', 'tags'];
```

Comment:
So instead of having to define these tags in two separate places what about just storing the whole object, and then below when you create the metadata you just assign the properties from the existing metadata?

`let metadata = Object.create(this._existingMetadata) as nb.INotebookMetadata;`